### PR TITLE
fix: download site bundle correctly if site name has special chars

### PR DESCRIPTION
**Pull request by pulkit0555**
### What does this PR do?
Fixes the bug when site name contains special chars and we need to download site bundle
### What issues does this PR fix or reference?
@W-16552899@

### CODE REVIEW

File: src/shared/experience/expSite.ts [hunk 1]
```diff
@@ -179,11 +179,21 @@ export class ExperienceSite {
...
+      const updatedSiteName = replaceSpacesAndSpecialChars(this.siteName);
+      this.siteName = updatedSiteName;
+    }
     const result = await this.org
       .getConnection()
       .query<{ Name: string; LastModifiedDate: string }>(
         `SELECT Name, LastModifiedDate FROM StaticResource WHERE Name LIKE 'MRT_experience_%_${this.siteName}'`
       );
+    // Changing the site name back to original after fetching data from StaticResource so it may not break the flow elsewhere
+    // this.siteName = originalSiteName;
```
Comment (pulkit0555) [inline src/shared/experience/expSite.ts:195]: @nrkruk If we uncomment this, it breaks the flow when we try to bring the site up in some cases where site name have special chars other than "-"

File: src/shared/experience/expSite.ts [hunk 2]
```diff
@@ -356,3 +366,27 @@ function getSiteNameFromStaticResource(staticResourceName: string): string {
   }
   return parts.slice(4).join(' ');
 }
+
+export function replaceSpacesAndSpecialChars(value: string): string {
+  // Replace any special characters with underscore
+  value = value.replace(/[^a-zA-Z0-9]/g, '_');
```
[the remark this answers is not in the document]
Comment (nrkruk) [reply src/shared/experience/expSite.ts:372]: Is that true of how the site names are saved in the static resources?

File: src/shared/experience/expSite.ts (hunk 1 above)
Comment (nrkruk) [reply src/shared/experience/expSite.ts:195]: Not sure what this means. Does this impact functionality in some way?

**Comment by nrkruk** [comment]
We will also need to backport this to the 252-patch branch

File: src/shared/experience/expSite.ts (hunk 1 above)
Comment (pulkit0555) [reply src/shared/experience/expSite.ts:195]: @nrkruk  I was resetting the site name back to original after executing the query but when it tries to bring up the site it fails. So, basically we will get that site bundle but it won't bring up the site.

Also, should we do the site name updation inside the constructor itself? It's already replacing the space with underscore character.

```
  public constructor(org: Org, siteName: string) {
    this.org = org;
    this.siteDisplayName = siteName.trim();
    this.siteName = this.siteDisplayName.replace(' ', '_');
  }
```

File: src/shared/experience/expSite.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pulkit0555) [reply src/shared/experience/expSite.ts:372]: I have tested with below examples of site names, works fine.

- space separated site
- site@with#special-chars
- site-with-hyphen
- byoNoBasePath
- byo enhanced

**Comment by pulkit0555** [comment]
Closing this forked repo PR. Will create a new one.

**Pull request closed without merging**